src/Components/FirmwareSelector/index.jsx

```diff
@@ -55,25 +54,27 @@ function FirmwareSelector({
 
   const file = useRef(null);
 
-  // Pre select ESC if escHint is a valid layout
+  // Pre select current firmware and ESC layout if valid
   useEffect(async () => {
     const availableFirmware = Object.keys(escs);
-    const validSources = getSupportedSources(signatureHint);
+    const validSources = getSupportedSources(esc && esc.meta.signature);
```

Comment:
esc check can be omitted if we have a proper default set in the prop validation.